### ISSUE TITLE
Concept: transformer for file layouts

### ISSUE THREAD

**Issue by zkamvar** [carpentries/sandpaper#69]
One of the things I want to be aware of is the fact that we may want to switch out the engines on the back end since we are at the moment shoe-horning {pkgdown} to do our dirty work for us, but different engines expect different file layouts (not to mention template languages).

Allison Hill and Desirée De Leon did a really good job of describing this problem in [their talk on Sharing on Short Notice](https://rstudio-education.github.io/sharing-short-notice/). 

It would be nice to be able to say "I want to take all the sources (and outputs) from my Jekyll site and migrate them to my Hugo site" or "Hugo -> Hugo" or "Hugo -> Distill" and run a single function that does it magically.

**Basically, I want to take one complex nested structure and transform it to another complex nested structure.**

The closest thing I can get to this concept is: https://github.com/mcmath/deep-map

It may be worthwhile exploring this concept because I can definitely see it being applicable to not just file paths, but also metadata structures (which will eventually be in our scope!)
